### PR TITLE
SQMAGOPC-491: Pay and add card - functionality

### PR DESCRIPTION
**Pull request by bartoszkaluzny-solteq**

### CODE REVIEW

File: Gateway/Http/Client/TransactionPayAndAddCard.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        } catch (RequestException $e) {
+            $this->log->error(\sprintf(
+                'Connection error to Paytrail Payment Service API: %s Error Code: %s',
+                $e->getMessage(),
+                $e->getCode()
+            ));
+
+            if ($e->hasResponse()) {
+                $response["error"] = $e->getMessage();
+                return $response;
```
[the remark this answers is not in the document]
Comment (bartoszkaluzny-solteq) [reply Gateway/Http/Client/TransactionPayAndAddCard.php:106]: Removed

File: Controller/Callback/PayAndAddCardCallback.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+     */
+    private function saveToken($params, $order)
+    {
+        $vaultPaymentToken = $this->paymentTokenFactory->create(PaymentTokenFactory::TOKEN_TYPE_CREDIT_CARD);
+        $customerId = $order->getCustomerId();
+        $vaultPaymentToken->setCustomerId($customerId);
+        $vaultPaymentToken->setPaymentMethodCode($this->gatewayConfig->getCcVaultCode());
+        $vaultPaymentToken->setExpiresAt(
+            sprintf(
+                '%s-%s-01 00:00:00',
```
[the remark this answers is not in the document]
Comment (bartoszkaluzny-solteq) [reply Controller/Callback/PayAndAddCardCallback.php]: image
True. I changed this to 23:59:59

[the remark this answers is not in the document]
Comment (masloorzechowe) [reply Controller/Callback/PayAndAddCardCallback.php]: but also last day of the month
so need to check every month when saving token, to set proper date

File: Controller/Callback/PayAndAddCardCallback.php [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                $params['expire_year'],
+                bcadd($params['expire_month'], '1')
```
[the remark this answers is not in the document]
Comment (bartoszkaluzny-solteq) [reply Controller/Callback/PayAndAddCardCallback.php]: Yes found it out 😄 I changed calculations, now it should be ok. 
`date("Y-m-t 23:59:59", strtotime($expiresDate))` This returns last day of month and time 23:59:59

**Pull request merged**